earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/shared/glc_unpacker.cpp

```diff
@@ -448,6 +448,30 @@ bool GlcUnpacker::FindFile(const char* file_name,
   }
 }
 
+/**
+ * Find each file in a provided directory on a layer and retrieve their names,
+ * offsets, and sizes.
+ */
+bool GlcUnpacker::FindLayerDirectoryFiles(const char* directory_name,
+                                     int layer,
+                                     std::map<const char* , PackageFileLoc*>* file_locs) {
+    FileUnpacker* unpacker = GetUnpacker(layer);
+
+    for (int i = 0; i < unpacker->IndexSize(); ++i) {
+      const char *package_file = unpacker->IndexFile(i);
+
+      PackageFileLoc file_loc;
+
+      if (FindFile(package_file, &file_loc)) {
+        if (!strncmp(directory_name, package_file, strlen(directory_name))) {
+          (*file_locs).emplace(package_file, &file_loc);
```

Comment:
you are passing the caller a pointer to a stack allocated variable that will likely be invalid after this call (or even loop iteration).  I might think a walker would be more flexible.  Then the caller can choose to either use the data right way (avoiding the copy) or save the data for use later (then likely needing to do a copy) or a mix there of.

Reply:
See `MapDataPacketWalker` in this same class.  Maybe have a file walker where you can walk the embedded files of a portable file and do things with them as they are walked.